From: Yongjian Xu <xuyongjiande@gmail.com>
Date: Tue, 30 Dec 2014 16:03:46 +0800
Subject: [PATCH 44/58] qlcnic: Fix return value in qlcnic_probe()
Patch-mainline: v3.19-rc4
Git-commit: 7824acd92494cf21229ea5313e525fa20927ba26
References: bsc#922064 FATE#318609

If the check of adapter fails and goes into the 'else' branch, the
return value 'err' should not still be zero.

Signed-off-by: Yongjian Xu <xuyongjiande@gmail.com>
Acked-by: Shahed Shaikh <shahed.shaikh@qlogic.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Gary Ching-Pang Lin <glin@suse.com>
---
 drivers/net/ethernet/qlogic/qlcnic/qlcnic_main.c |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_main.c
+++ b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_main.c
@@ -2591,6 +2591,7 @@ qlcnic_probe(struct pci_dev *pdev, const
 	} else {
 		dev_err(&pdev->dev,
 			"%s: failed. Please Reboot\n", __func__);
+		err = -ENODEV;
 		goto err_out_free_hw;
 	}
 
